datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -907,8 +907,12 @@ public String getDatastreamTaskStateForKey(DatastreamTask datastreamTask, String
   public void setDatastreamTaskStateForKey(DatastreamTask datastreamTask, String key, String value) {
     String path = KeyBuilder.datastreamTaskStateKey(_cluster, datastreamTask.getConnectorType(),
         datastreamTask.getDatastreamTaskName(), key);
-    _zkclient.ensurePath(path);
-    _zkclient.writeData(path, value);
+    String taskPath = KeyBuilder.connectorTask(_cluster, datastreamTask.getConnectorType(),
+        datastreamTask.getDatastreamTaskName());
+    if (_zkclient.exists(taskPath)) {
+      _zkclient.ensurePath(path);
```

Comment:
I'm guessing this ensurePath() is necessary just to save us from race conditions, where the node gets deleted after we check for existence?